core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -922,6 +922,24 @@ private[spark] object Utils extends Logging {
   }
 
   /**
+   * Wrapper over newSingleThreadExecutor. Thread names are formatted as prefix-ID, where ID is a
+   * unique, sequentially assigned integer.
+   */
+  def newDaemonSingleThreadExecutor(prefix: String): ExecutorService = {
```

Comment:
probably as a separate PR - might be good to create a ThreadUtils.scala and move all the thread pool/executor stuff there.